File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/csharp/src/de/sots/cellarsandcentaurs/domain/model/NoSuchCreatureException.cs
using System;

namespace de.sots.cellarsandcentaurs.domain.model
{
    public class NoSuchCreatureException : Exception
    {
        public NoSuchCreatureException(CreatureId creatureId)
            : base($"No such creature with ID: {creatureId}")
        {
        }
    }
}

File: training/javascript/src/de/sots/cellarsandcentaurs/domain/model/NoSuchCreatureException.js
import { CreatureId } from "./CreatureId.js";

export class NoSuchCreatureException extends Error {
    /**
     * @param {CreatureId} id
     */
    constructor(id) {
        super("No such creature in the dungeon: " + id.id);
        this.name = "NoSuchCreatureException";
    }
}
